doc/embedded/crosscompile_example.md

```diff
@@ -0,0 +1,194 @@
+## Crosscompile mlpack example for an embedded hardware
+
+In this article, we explore how to crosscompile and run an mlpack example code
+on an embedded hardware. In our previous
+[documentations](crosscompile_armv7.cmake), we have explored how to run mlpack
+bindings such as kNN command line program on a Raspberry PI 2. Please refer to that
+article first and follow the first part on how to setup cross-compilation toolchain,
+and then continue with this article. Or refer to this
+[table](supported_boards.md) for a quick start guide.
+
+### Cloning mlpack example respository
+
+mlpack has an [example repository](https://github.com/mlpack/examples) that
+demonstrates a set of examples showing how to use the library source code
+on different dataset and usecases including embedded deployment. This
+tutorial basically explains our necessary CMake configurations
+that are required to integrate with your local CMake to download mlpack
+dependencies and cross compile the entire software.
+
+If you have not used mlpack example repsoitory, I highly recommend to clone
+this repository from the following link:
+
+```sh
+git clone git@github.com:mlpack/examples.git
+```
+
+You can explore this repository and see the available examples. However, in this
+tutorial, we are interested in the `embedded/` directory. In this directory we are providing
+a CMake template project and a RandomForest example to use it on embedded
+hardware. For now, we are exploring the RandomForest example by 
+analysing the CMakeLists.txt.
+
+### Analysing CMakeLists.txt
+
+In this part of code we are defining the project name and including two CMake
+configurations files. The first one is the `Autodowload.cmake` function that is going
+to pull the dependencies from respective locations that we are going to define.
+
+The second configuration file is `ConfigureCrossCompile.cmake`. This will allow
+to include all the necessary configurations to allow using a cross compiler
+toolchain.  Then we are setting the C++ standard to use and define settings
+for `OpenMP` and `std::thread`
+
+```cmake
+cmake_minimum_required(VERSION 3.6)
+project(RandomForest)
+
+include(CMake/Autodownload.cmake)
+include(CMake/ConfigureCrossCompile.cmake)
+
+option(USE_OPENMP "If available, use OpenMP for parallelization." ON)
+
+# Set required standard to C++17.
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# If we're using gcc, then we need to link against pthreads to use std::thread,
+# which we do in the tests.
+if (CMAKE_COMPILER_IS_GNUCC)
+  find_package(Threads)
+  set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+endif()
+```
+
+#### Downloading dependencies
+
+Once we have included the configs and defined the C++ standard, then we need to
+download these dependencies. In the following, we have added the link
+for each one of them, but feel free to adapt the link to different versions or
+locations.
```

Comment:
```suggestion
The next steps are to use the functionality of the autodownloader to actually
download mlpack's dependencies.
```

I think we can simplify this paragraph.  But... when I got to here, I thought: is this really the right way to present this application?

In the "best" situation, a user adds

```
find_package(mlpack)
```

to their CMake configuration (and maybe puts a `Findmlpack.cmake` file in place---we have one of those in the models repository), and this tries to find mlpack and all its dependencies locally.

If the user doesn't have mlpack's dependencies, we can also point them to the autodownloader to download everything---but we should also extend autodownloader support to mlpack itself.